apps/frontend/static_src/js/main.js

```diff
@@ -1 +1,70 @@
 import { Dropdown } from 'bootstrap';
+
+const searchInput = document.querySelector('[data-search-input]')
+const searchIconButton = document.querySelector('[data-search-icon-button]')
+
+const onSearchInputChange = async (event) => {
+    const query = event.target.value
+    try {
+        const res = await fetch(window.location.origin + '/search_json/?query=' + query)
+        const data = await res.json()
+        injectResultsInHTML(data)
+    } catch(err) {
+        console.log(err)
+        return {"error": err}
+    }
+}
+searchInput.addEventListener("keyup", (event) => {
+    setTimeout(() => {
+        onSearchInputChange(event)
+    }, 1000)
+})
+
+
+const removeExistingChildren = (parent) => {
+    const children = [...parent.children]
+    for (const child of children) {
+        child.remove()
+    }
+}
+searchIconButton.addEventListener("click", () => {
+    const resultsDiv = document.querySelector('[data-results]')
+    const resultsCountContainer = document.querySelector('[data-results-count-container]')
+
+    removeExistingChildren(resultsDiv)
+    removeExistingChildren(resultsCountContainer)
+})
+
+const injectResultsInHTML = (results) => {
+    const resultsDiv = document.querySelector('[data-results]')
+    const resultsCountContainer = document.querySelector('[data-results-count-container]')
+
+    removeExistingChildren(resultsDiv)
+    removeExistingChildren(resultsCountContainer)
+
+    const resultsCountDiv = document.createElement("div")
+    const resultsCountData = document.createTextNode(`${results.length} ${results.length == 1 ? 'result' : 'results'} found`)
+    resultsCountDiv.appendChild(resultsCountData)
+    resultsCountDiv.classList.add('m-3', 'mx-5')
+    resultsCountContainer.appendChild(resultsCountDiv)
+
+    for(const result of results) {
+        const resultDiv = document.createElement("div")
+        const resultLink = document.createElement("a")
+        const resultDescription = document.createElement("div")
+        const resultParentSection = document.createElement("div")
+        resultLink.appendChild(document.createTextNode(result.title))
+        resultDescription.appendChild(document.createTextNode(result.search_description))
+        resultParentSection.appendChild(document.createTextNode(result.parent_section))
+        resultLink.href = result.url_path.replace('home-x/', '').replace('home-x-', '')
```

Comment:
This looks a bit odd to me. I would expect something like this to happen server-side, or if we want to keep it here, then at least add a comment.

Reply:
This was something confusing to me as well and I wanted to ask about this but it just slipped out of my mind :sweat_smile: .
Here we have a home page with slug `home-x`. However on the site the slug isn't used for home or root page (as far as I know). But in the search results the slug is present.
So `/` would be returned as `/home-x`, `/tutorial/` as `/home-x/tutorial/` and so on.

Reply:
As a result when you click on any of the search-results page, you would get 404.